File: 자료구조/큐/빽준_10845_큐/큐.py
from sys import stdin

n = int(stdin.readline())
arr = [0 for _ in range(10001)]
f = 0
b = 0
while n > 0:
    n-=1
    str = list(stdin.readline().split())
    if str[0] == "push":
        arr[b] = int(str[1])
        b+=1
    elif str[0] == "pop":
        if f == b: print("-1")
        else:
            print(arr[f])
            arr[f] = 0
            f+=1
    elif str[0] == "size": print(b-f)
    elif str[0] == "empty":
        if f == b: print("1")
        else: print("0")
    elif str[0] == "front":
        if f == b: print("-1")
        else: print(arr[f])
    elif str[0] == "back":
        if f == b: print("-1")
        else: print(arr[b-1])

File: 자료구조/큐/빽준_10845_큐/큐.cs
using System;
using System.Text;
using static System.Console;

class Program
{
    public static void Main(string[] args)
    {
        StringBuilder ans = new StringBuilder();
        int[] arr = new int[100001];
        int n = int.Parse(ReadLine());
        int f = 0;
        int b = 0;
        while (n-- > 0)
        {
            string[] s = ReadLine().Split(' ');
            if (s[0] == "push")
            {
                int num = int.Parse(s[1]);
                arr[b] = num;
                b++;
            }
            else if (s[0] == "pop")
            {
                if (f == b) ans.Append(-1 + "\n");
                else
                {
                    ans.Append(arr[f] + "\n");
                    arr[f] = 0;
                    f++;
                }
            }
            else if (s[0] == "size") ans.Append(b-f + "\n");
            else if (s[0] == "empty")
            {
                if(f == b) ans.Append(1 + "\n");
                else ans.Append(0 + "\n");
            }
            else if (s[0] == "front")
            {
                if(f == b) ans.Append(-1 + "\n");
                else ans.Append(arr[f] + "\n");
            }
            else if (s[0] == "back")
            {
                if(f == b) ans.Append(-1 + "\n");
                else ans.Append(arr[b-1] + "\n");
            }
        }
        WriteLine(ans.ToString());
    }
}